woocommerce/woocommerce-gateway-stripe
# Fixes `_doing_it_wrong()` messages being displayed on WordPress 6.7 due to translating too early

Fixes #3604

Related P2: peNR48-111-p2

## Changes proposed in this Pull Request:







After updating to WordPress 6.7, stores using our Stripe extension and have `WP_Debug` enabled will start seeing the following `_doing_it_wrong` notices across WP Admin pages:

![image](https://github.com/user-attachments/assets/841b2db8-37a0-46b6-b38f-2be306dd52ce)

In WordPress 6.7, they [changed the way translations are loaded](https://core.trac.wordpress.org/changeset/59127) which affects how and when translation strings should be triggered. With this change, any `__()` code called before the `after_setup_theme` is considered "too early" and will need to be updated to happen on a later hook.

Part of the challenge with Stripe is that the entire plugin is inited on the `'plugins_loaded'` hook (which is before `after_setup_theme`):

```php
add_action( 'plugins_loaded', 'woocommerce_gateway_stripe_init' );
```

Digging through all of the files and classes included/initiated `WC_Stripe->init()` I found the following issues within Stripe:

- `WC_Stripe_Privacy` translated strings in the `__construct()`. Fixed in a4b1be2aa5f85f54be23c436f06439c2cd29d81d
- `WC_Stripe_Settings_Controller` was being instantiated with passing `get_main_stripe_gateway()` as the second parameter. This function builds the main Stripe gateway class which translates a bunch of strings. 5b88b55c8ce87a5e88a6891a66ffa7cb1ad7d102
- `load_plugin_textdomain()` was being called on `plugins_loaded`, this should happen on init 0361256bf4e7d9475cc021ad82c254777ee6b0a8

## Testing instructions





To confirm the bug:
1. Make sure `define( 'WP_DEBUG', true );` is set in wp-config
2. Update to the latest WordPress 6.7
3. Go to your Edit User Profile page and update the language to Español
![image](https://github.com/user-attachments/assets/de67b3ce-0184-444f-8bad-75a95989f433)

There are two key areas of Stripe that these changes impact:

#### 1. Privacy erasers/exporters

1. Go to WP Admin > Tools > Export personal data
5. Enter your email and manually download your personal data
6. Download and open zip -> index.html file
7. Under Customer Data heading, confirm "Stripe payment id" exists

#### 2. Settings Controller

1. Go to WooCommerce > Settings > Payments > Stripe
2. Confirm no issues with loading the settings

---


**Post merge**
